### DISCUSSION TITLE
What is parameter --upper-level when download data?

### DISCUSSION THREAD

**Discussion by dzungpv** [open-meteo/open-meteo#425] [Q&A] [answered]
Iam optimizing self host API server and want to reduce resource for the server, what does the --upper-level parameter when downloading the data? If i remove it what data is missing from the models?

**Comment by patrick-zippenfenig** [comment, marked as answer]
Hi,
adding `--upper-level` to the downloaders will download alle pressure level variables from a given weather model. Upper level variables require significantly more storage than regular surface variables. If you do not need them, do not download them. You can also specify a list of weather variables you want to download with `--only-variables`

**Reply by dzungpv** [reply]
If I want precipitation probability, what data should I download? Because now with my settings it is missing.

**Reply by patrick-zippenfenig** [reply]
It's downloaded by `openmeteo-api download-gfs gfs025_ensemble`. However, this will change in the future, as additional models have to be considered for precipitation probability

**Reply by dzungpv** [reply]
Thank you, gfs025_ensemble not include in the docs, So I try gfs025_ens it can produce precipitation probability, but it is 21GB, too big. You save my server.
